### PR TITLE
Nvm/update node version

### PR DESCRIPTION
**Pull request by TechDufus**
TLDR: Create an update / upgrade path for nvm/node versions without borking nodes. This will allow upgrades / downgrades to the `nvm.node.version` provided in the `vars.yml` file.

This pull request includes changes to the roles and variables related to NVM, Node, Logrotate, PM2, and Watchdog. The most important changes include adding tasks to check and install NVM and Node, installing Logrotate, PM2, and Watchdog, and updating the Node version in the `vars.yml` file.

Summary of changes:

* <a href="diffhunk://#diff-c4f5a5bb8c501f04872e493e2f2366c5ad175921b5c0d4ab633bb9049e254679L2-R48">`roles/nvm/tasks/nvm-install.yml`</a>: Added tasks to check if NVM and Node are installed, get the installed Node version, determine if a Node update is needed, install Logrotate, PM2, and Watchdog, and start PM2 after installation. <a href="diffhunk://#diff-c4f5a5bb8c501f04872e493e2f2366c5ad175921b5c0d4ab633bb9049e254679L2-R48">[1]</a> <a href="diffhunk://#diff-c4f5a5bb8c501f04872e493e2f2366c5ad175921b5c0d4ab633bb9049e254679R93-R140">[2]</a>
* <a href="diffhunk://#diff-8dfad57c9c5569359b073d71d8a8f0503cca20a8f657ac7137db2fc7e504a2a4R12-R41">`roles/nvm/tasks/nvm-purge.yml`</a>: Added tasks to purge Logrotate, PM2, and Watchdog with specific tags and elevated privileges.
* <a href="diffhunk://#diff-5580739af52e3b23a2c760d9568fde1cf6b6b794f40b7f2635f0f5f18aec8d78R9">`roles/pm2/tasks/pm2-stop.yml`</a>: Added the `ignore_errors` option to the `ansible.builtin.shell` task to allow it to continue executing even if it encounters an error.
* <a href="diffhunk://#diff-f149cfca468975e4755d447af3239298fc0ce15a4c3015c469823073c211f9f2L32-R32">`vars.yml`</a>: Updated the Node version from `16.19.1` to `20.9.0` to specify the desired version of Node to be installed.

### CODE REVIEW

**Comment by TechDufus** [comment]
I marked as draft, because if merged as-is, this will require a full node and nvm purge in order to update node (which will need to reinstall nvm). If we break out the node install steps into it's own role `node`, we can easily uninstall node and reinstall the desired version without having to reinstall `nvm` itself.

I will work on breaking the current `nvm` role into --> `nvm` & `node` roles.
